transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java

```diff
@@ -80,14 +83,16 @@ public void initialize(TransformContext context) throws Exception {
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
     pipelineConfigurer.getStageConfigurer().setOutputSchema(outputSchema);
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
 
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (inputSchema.getField(config.inputField) == null) {
-        throw new IllegalArgumentException(String.format("Field %s is not present in input schema", config.inputField));
-      }
-      if (inputSchema.getField(config.inputField).getSchema().getType() != Schema.Type.STRING) {
-        throw new IllegalArgumentException(String.format("Field %s must be of type string.", config.inputField));
+        collector.addFailure(String.format("Field '%s' must be present in input schema.", config.inputField), null)
+            .withConfigProperty(INPUT_FIELD);
+      } else if (inputSchema.getField(config.inputField).getSchema().getType() != Schema.Type.STRING) {
```

Comment:
this is overly verbose, it's easier to read if it's like

```
Schema.Field field = inputSchema.getField(config.inputField);
if (field == null) {
  ...
} else if (...
```

Reply:
Fixed